etl/extractor/platform_impl.go

```diff
@@ -211,34 +212,105 @@ func closeStream(ctx context.Context, stream exporter.RecordExporter_ExportClien
 	}
 }
 
-func (m *PlatformExtractor) Stop(ctx context.Context) error {
-	m.startStopMutex.Lock()
-	defer m.startStopMutex.Unlock()
-	if m.hasStarted {
+func (e *PlatformExtractor) Stop(ctx context.Context) error {
+	e.cancel()
+	e.startStopMutex.Lock()
+	defer e.startStopMutex.Unlock()
+	if e.hasStarted {
 		belogger.FromContext(ctx).Info("Stopping platform extractor...")
-		m.stopSignal <- true
-		m.hasStarted = false
+		e.stopSignal <- true
+		e.hasStarted = false
 	}
 	return nil
 }
 
-func (m *PlatformExtractor) needStop() bool {
+func (e *PlatformExtractor) needStop() bool {
 	select {
-	case <-m.stopSignal:
+	case <-e.stopSignal:
 		return true
 	default:
 		// continue
 	}
 	return false
 }
 
-func (m *PlatformExtractor) Start(ctx context.Context) error {
-	m.startStopMutex.Lock()
-	defer m.startStopMutex.Unlock()
-	if !m.hasStarted {
+func (e *PlatformExtractor) Start(ctx context.Context) error {
+	e.startStopMutex.Lock()
+	defer e.startStopMutex.Unlock()
+	if !e.hasStarted {
 		belogger.FromContext(ctx).Info("Starting platform extractor...")
-		m.getJetDropsContinuously(ctx)
-		m.hasStarted = true
+		// e.getJetDropsContinuously(ctx)
+		e.hasStarted = true
+		ctx, e.cancel = context.WithCancel(ctx)
+		go e.retrievePulses(ctx)
 	}
 	return nil
 }
+
+func (e *PlatformExtractor) retrievePulses(ctx context.Context) {
+	pu := new(exporter.FullPulse)
+	var err error
+	logger := belogger.FromContext(ctx)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+		time.Sleep(5 * time.Second)
+		before := pu.PulseNumber
+		pu, err = e.pulseExtractor.GetNextFinalizedPulse(ctx, int64(before))
+		log := logger.WithField("pulse_number", pu.PulseNumber)
+		if err != nil {
+			log.Error("GetNextFinalizedPulse(): ", err)
+			pu.PulseNumber = 0
+			continue
+		}
+		if pu.PulseNumber == before {
+			continue
+		}
+
+		log.Info("retrieved")
```

Comment:
let's change to "retrieved new pulse number"

Reply:
ok